server/src/main/java/org/eclipse/openvsx/web/WebConfig.java

```diff
@@ -46,6 +46,8 @@ public void addCorsMappings(CorsRegistry registry) {
                     .allowedOrigins("*");
             registry.addMapping("/api/**")
                     .allowedOrigins("*");
+            registry.addMapping("/vscode/gallery/extensionquery")
```

Comment:
Thanks for the contribution! Shall we extend this to all endpoints in the vscode adapter? That would also cover the endpoints for fetching files.
```suggestion
            registry.addMapping("/vscode/**")
```

Reply:
Probably a good idea. I didn't look to see what other endpoints might need covered, only addressed what was in the bug.

Upon closer inspection it looks like there are potentially other endpoints also not covered by the @CrossOrigin annotation. These are:

/vscode/item
/vscode/gallery/publishers/{namespace}/vsextensions/{extension}/{version}/vspackage

I've added a new commit covering your suggested change, as well as adding @CrossOrigin to the two other endpoints under /vscode/.

Please note that it seems there is also an infrastructure issue blocking CORS headers from being transmitted from the production deployment to clients, as noted by daiyam.